src/full_client.toit

```diff
@@ -489,29 +496,34 @@ class MemoryPersistenceStore implements PersistenceStore:
   */
   storage_ /Map := {:}
 
-  add topic/string payload/ByteArray --retain/bool -> int:
-    return add (PersistedPacket topic payload --retain=retain)
-
-  add packet/PersistedPacket -> int:
-    id := ?
-    if storage_.is_empty: id = 0
-    else: id = storage_.last + 1
+  add packet/PersistedPacket -> none:
+    id := packet.packet_id
     storage_[id] = packet
-    return id
 
-  get persistent_id/int -> PersistedPacket?:
-    return storage_.get persistent_id
+  get persisted_id/int -> PersistedPacket?:
+    return storage_.get persisted_id
 
-  remove persistent_id/int -> none:
-    storage_.remove persistent_id
+  remove_persisted_with_id persisted_id/int -> bool:
+    result := storage_.contains persisted_id
+    storage_.remove persisted_id
```

Comment:
Maybe use `if_absent` parameter to return false? Seems better than having to hash and lookup twice.

Reply:
Forgot that we had an `--if_absent`. Nice.
done.